go.mod

```diff
@@ -3,14 +3,15 @@ module github.com/thanos-io/objstore
 go 1.20
 
 require (
-	cloud.google.com/go/storage v1.35.1
+	cloud.google.com/go/storage v1.40.0
```

Comment:
Do we need to update the library? Maybe we can add this to CHANGELOG.

Reply:
This got pulled in because of peer libraries (gcsemu). But downstream users still can pin their versions.